src/tests/ftest/io/ior_intercept_basic.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+from ior_test_base import IorTestBase
+from ior_utils import IorCommand, IorMetrics
+
+
+class IorIntercept(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Test class Description: Runs IOR with and without interception
+       library on a single server and single client setting with
+       basic parameters.
+
+    :avocado: recursive
+    """
+
+    def test_ior_intercept(self):
+        """Jira ID: DAOS-3498.
+
+        Test Description:
+            Purpose of this test is to run ior using dfuse for 5 minutes
+            and capture the metrics and use the intercepiton library by
+            exporting LD_PRELOAD to the libioil.so path and rerun the
+            above ior and capture the metrics and compare the
+            performance difference and check using interception
+            library make significant performance improvement.
+
+        Use case:
+            Run ior with read, write, CheckWrite, CheckRead
+                for 5 minutes
+            Run ior with read, write, CheckWrite, CheckRead
+                for 5 minutes with interception library
+            Compare the results and check whether using interception
+                library provides better performance.
+
+        :avocado: tags=all,daosio,medium,hw,full_regression,iorinterceptbasic
+        """
+        out = self.run_ior_with_pool()
+        without_intercept = IorCommand.get_ior_metrics(out)
+        intercept = self.prefix + "/lib64/libioil.so"
```

Comment:
This should be os.path.prefix()

Reply:
I believe the self.prefix already contains the path to the build and may be I should use os.path.join() to concatenate lib64 and libioil.so.

> What is this checking for? This looks like it runs IOR with a pool (I'm not sure about a container) and logs the results. Given that the interception library will silently fail back to dfuse if there's an error this would only catch catastrophic errors in the intersection library, not errors which prevent it from functioning.

Yes, this test runs IOR on a DFUSE mount before and after loading the libioil.so and compares the results to check the performance. It's using the existing ior run framework to setup DFUSE and container creation. What's the good/reliable way to check if the interception library failed/fall back to dfuse and what kind of things would fail it ? 
Other than looking at the ior results I haven't found any other way what's going on with interception library. So, with this assertion, the assumption is that if the performance number is not as expected there's something wrong and the test would fail. 
May be I could check if the ior numbers are close to the ones without using interception library, I can assume the interception library fall back to dfuse.